src/main/webapp/app/shared/virtual-scroll/virtual-scroll.component.ts

```diff
@@ -0,0 +1,255 @@
+/**
+ * Component based on Open Source Project ag-virtual-scroll
+ * https://github.com/ericferreira1992/ag-virtual-scroll
+ *
+ */
+
+import { Component, ElementRef, EventEmitter, Input, OnChanges, OnDestroy, OnInit, Output, Renderer2, SimpleChanges, ViewChild } from '@angular/core';
+import { NavigationStart, Router } from '@angular/router';
+import { VirtualScrollRenderEvent } from 'app/shared/virtual-scroll/virtual-scroll-render-event.class';
+import { ITEMS_PER_PAGE } from 'app/shared/constants/pagination.constants';
+
+@Component({
+    selector: 'virtual-scroll',
+    templateUrl: './virtual-scroll.component.html',
+    styleUrls: ['virtual-scroll.style.css'],
+})
+export class VirtualScrollComponent implements OnInit, OnChanges, OnDestroy {
+    @ViewChild('itemsContainer', { static: true }) private itemsContainerElRef: ElementRef<HTMLElement>;
+
+    @Input('items') public originalItems: any[] | undefined = [];
+    @Input() forceReload: boolean;
+    @Output() forceReloadChange = new EventEmitter<boolean>();
+    @Output() private onItemsRender = new EventEmitter<VirtualScrollRenderEvent<any>>();
+
+    minRowHeight = 126.7;
```

Comment:
that's a very specific value, why? looks like a magic number

Reply:
Yes, this is the minimum height a post component can occupy within the screen.

We need to cache the heights of previous posts to be able to scroll upwards toward them and render them smoothly. On occasions when a new item has arrived on the upper side of the list, it is not rendered before, and therefore we can't obtain its height. To keep distortions as minimum as possible in such scenario, we need this magic number, which is specific for post elements.

Reply:
Then store it as a constant with a descriptive name and set it from outside to increase the reusability of this component

Reply:
Done ✔️